File: python/gtk4/oop/application-window.py
import gi, sys
gi.require_version("Gtk", "4.0")
from gi.repository import Gtk

class App(Gtk.Application):
    APP_ID = "io.github.Miqueas.GTK-Examples.Python.Gtk4.OOP.Application"
    APP_TITLE = "Gtk.ApplicationWindow"

    def __init__(self):
        super().__init__(application_id = self.APP_ID)

    def do_activate(self):
        window = self.get_active_window()
        if window: window.present()

    def do_startup(self):
        # Ensures chain-up
        Gtk.Application.do_startup(self)
        
        window = Gtk.ApplicationWindow(application = self)
        window.set_title(self.APP_TITLE)
        window.set_default_size(400, 400)

if __name__ == "__main__":
    sys.exit(App().run(sys.argv))

File: cpp/gtk4/oop/application-window.cpp
#include <gtkmm.h>
#include <iostream>

const Glib::ustring APP_ID = "io.github.Miqueas.GTK-Examples.C.Gtk4.OOP.ApplicationWindow";
const Glib::ustring APP_TITLE = "Gtk::ApplicationWindow";

class App : public Gtk::Application {
  public:
    static Glib::RefPtr<App> create() {
      return Glib::make_refptr_for_instance<App>(new App());
    }

  protected:
    App() : Gtk::Application(APP_ID) {}

    void on_startup() override {
      // Ensures chain-up
      Gtk::Application::on_startup();

      auto window = Gtk::make_managed<Gtk::ApplicationWindow>();
      window->set_title(APP_TITLE);
      window->set_default_size(400, 400);
      add_window(*window);
    }

    void on_activate() override {
      Gtk::Window* window = get_active_window();
      if (window != nullptr) window->present();
    }
};

int main(int argc, char** argv) {
  return App::create()->run(argc, argv);
}